samples/embedded-map/src/components/EmbeddedMap/EmbeddedMapModel.ts

```diff
@@ -136,37 +219,112 @@ export default class EmbeddedMapModel extends ComponentModelBase {
+            return;
+        }
+
+        const url = `${this.imageQueryUrl}?client_id=${this.mapillaryKey}&closeto=${longitude},${latitude}&radius=${this.searchRadius}`;
+
+        const response = await fetch(url, {
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json",
+            },
+        });
+
+        const data = await response.json();
+
+        const imgkey = data?.features?.[0]?.properties?.key;
+
+        if (imgkey) {
+            await this.mapillary.moveToKey(imgkey);
+            this._currentPosition = this.mapillary.getPosition();
+            this._updating = false;
+        } else {
+            this._activateCover();
+        }
+    }
+
     // Throttle the updates to the Map to avoid overwhelming with pan/zoom
     // commands.
     private _onPerspectiveChange = throttle(async () => {
-        if (!this.map || !this.mapillary) {
+        if (!this.map || !this.mapillary || this._updating) {
             return;
         }
 
-        const [{ lat, lon }, bearing] = await Promise.all([
-            this.mapillary.getPosition(),
-            this.mapillary.getBearing(),
-        ]);
+        this._updating = true;
 
+        const {
+            latitude,
+            longitude,
+            heading,
+            tilt,
+            fov,
+        } = await this._getMapillaryCamera();
         const centerPoint = new Point({
-            latitude: lat,
-            longitude: lon,
+            latitude,
+            longitude,
         });
+
         await Promise.all([
             this.messages.commands.locationMarker.update.execute({
                 geometry: centerPoint,
-                heading: bearing,
+                heading,
+                tilt,
+                fov,
                 id: this.id,
                 maps: this.map,
             }),
-            this.messages.commands.map.zoomToViewpoint.execute({
-                maps: this.map,
-                viewpoint: {
-                    rotation: getCameraRotationFromBearing(bearing),
-                    targetGeometry: centerPoint,
-                    scale: 3000,
-                },
-            }),
-        ]);
+            this.syncGcxMap
+                ? this.messages.commands.map.zoomToViewpoint.execute({
+                      maps: this.map,
+                      viewpoint: {
+                          rotation: getCameraRotationFromBearing(heading),
+                          targetGeometry: centerPoint,
+                          scale: 3000,
+                      },
+                  })
+                : undefined,
+        ]).finally(() => (this._updating = false));
     }, 128);
+
+    // Gets the current POV of the mapillary camera
+    private async _getMapillaryCamera(): Promise<MapillaryCamera> {
+        if (!this.mapillary) {
+            return;
+        }
+
+        const [{ lat, lon }, { bearing, tilt }, fov] = await Promise.all([
+            this.mapillary.getPosition(),
```

Comment:
I discovered that `getPosition()` uses the camera position instead of the node position. Normally this probably wouldn't matter, but unfortunately it appears it makes our e2e tests very non-deterministic because of this.

I could see why `getPosition()` could be nice in some cases, as it will provide you the position as it does its position interpolation when moving from node to node.

https://github.com/mapillary/mapillary-js/blob/aa67afe85dacba0e738df2891b599d6b2371c510/src/viewer/Viewer.ts#L591-L606

If we don't care for the interpolation behavior, do you mind spending a bit of time seeing if it's easy for us to get node position instead? I believe this would make our tests much more reliable, and the difference shouldn't be perceivable to the end user (I hope). Or if you have any other ideas to change our tests to be more reliable

Reply:
The image objects do come with a position that is hopefully consistent, but I think this might not help, as we are syncing the viewer to the mapillary camera position, not the image position. So we'd end up checking the viewer position against the stated default image camera location and still finding that it didn't quite match. 

Another wrinkle is that mapillary could update their data at any time and return a different node for the same initial query and fail the test. Also, more than one mapillary image can be stored at the same location and we are assuming the returned results will be deterministic even when order is arbitrary.

Combine the above with the fact that setting the initial position from the web scene center rather than hardcoding it means this config must stay static too and the test is looking very fragile indeed. I've updated what remains of the test to pass at time of writing, but any of these things changing will cause it to fail again.

I can't really think of any solution to this right now other than mocking the web scene and accepting 'close enough' as an answer for the final marker location. The fact that we were able to query mapillary, get results, and then put a marker on the map in more or less the defined 'center' of the web scene does seem like it constitutes a fair test of this sample.

Reply:
Yeah agreed. However it may not be that bad. Checkout the [`nodechanged` event](https://mapillary.github.io/mapillary-js/classes/viewer.html#nodechanged). That should fire any time the current node changes, which should have the lat/lon attached (via [`Node.latLon`](https://mapillary.github.io/mapillary-js/classes/node.html#latlon))

In theory we should be able to hold on to the node as it changes, and use that as our source of truth instead of asking for the camera position via `getPosition()`. But of course we lose the interpolation (if we're okay with that). 

You should also be able to keep interpolation as well if you really wanted by leveraging `movestart` and `moveend` events. May end up making things more complicated than they need to be however.

With re. to fuzzy matching in the tests, i'm already currently doing that, but at some point it started to feel like the values were so different that i was losing confidence in the test.

Reply:
I think this might explain what we are seeing:
https://bl.ocks.org/oscarlorentzon/16946cb9eedfad2a64669cb1121e6c75

It looks like the nodechanged event may fire twice, once with the original GPS position and again when a computed location is derived. The computed location is the one we should be using as it is derived from a 3d reconstruction and will be much more accurate.

Reply:
Updated the code to account for the above and re-enabled the full test.